src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const oldVersion = {};
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    oldVersion[fromFormat[i]] = dateParts[i];
+
+    if (fromFormat[i] === 'YYYY' && toFormat.includes('YY')) {
+      oldVersion['YY'] = oldVersion['YYYY'].slice(2);
+    }
+
+    if (fromFormat[i] === 'YY' && toFormat.includes('YYYY')) {
+      if (oldVersion['YY'] < 30) {
+        oldVersion['YYYY'] = `20${oldVersion['YY']}`;
+      } else {
+        oldVersion['YYYY'] = `19${oldVersion['YY']}`;
+      }
```

Comment:
Similar to the previous comment, when converting from 'YY' to 'YYYY', there's no need to check if 'YYYY' is included in the 'toFormat'. You can directly perform the conversion based on the value of 'YY'.